Formula/p/progressline.rb

```diff
@@ -0,0 +1,22 @@
+class Progressline < Formula
+  desc "Track commands progress in a compact one-line format"
+  homepage "https://github.com/kattouf/ProgressLine"
+  url "https://github.com/kattouf/ProgressLine/archive/refs/tags/0.2.1.tar.gz"
+  sha256 "5ad7ff2b766f59b25f0e197e25b728e1272dc38fd3bd377480e8212f6e03abad"
+  license "MIT"
+  head "https://github.com/kattouf/ProgressLine.git", branch: "main"
+
+  depends_on xcode: ["15.4", :build]
+
+  uses_from_macos "swift"
+
+  def install
+    system "swift", "build", "--disable-sandbox", "--configuration", "release"
+    bin.install ".build/release/progressline"
+  end
+
+  test do
+    # NOTE: this is interactive tool that hard to test in release configuration
+    system bin/"progressline", "--help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Ok, thank you, I'll try some of this.

Reply:
Updated ✅